### PR TITLE
Add CSON primer

### PR DESCRIPTION
**Pull request by mnquintana**
Closes https://github.com/atom/docs/issues/43
Closes #102
Refs https://github.com/atom/atom/issues/2695#issuecomment-125680992

/cc @atom/issue-triage

### CODE REVIEW

**Comment by thomasjo** [comment]
Nice, I like it.

File: book/02-using-atom/sections/02-text.asc [hunk 1]
```diff
@@ -289,6 +289,8 @@ console.log("crash");
 
 The string `"crash"` would be initially selected and pressing tab again would place the cursor after the `;`
 
+WARNING: Snippet keys, unlike CSS selectors, can only be repeated once per level. If there are duplicate keys, then only the last one will be read. See <<_cson,Configuring with CSON>> for more information.
```
Comment (izuzak) [inline book/02-using-atom/sections/02-text.asc]: > If there are duplicate keys, ...

Perhaps "If there are duplicate keys **at the same level**..." just to be :gem: clear?

File: book/02-using-atom/sections/06-customizing.asc [hunk 2]
```diff
@@ -3,6 +3,21 @@
 
 Now that we are feeling comfortable with just about everything built into Atom, let's look at how to tweak it. Perhaps there is a keybinding that you use a lot but feels wrong or a color that isn't quite right for you. Atom is amazingly flexible, so let's go over some of the simpler flexes it can do.
 
+[[_cson]]
+==== Configuring with CSON
+All of Atom's config files (with the exception of your <<_style_tweaks, style sheet>> and your <<_the_init_file, Init Script>>) are written in CSON, short for CoffeeScript Object Notation.footnoteref:[cson, https://github.com/bevry/cson#what-is-cson] Just like its namesake JSON (JavaScript Object Notation), CSON is a text format for storing structured data in the form of simple objects made up of key-value pairs.
```
Comment (izuzak) [inline book/02-using-atom/sections/06-customizing.asc]: I'd probably :link: to JSON as well. :older_man: Always be linkin' :smile:

**Comment by izuzak** [comment]
:zap: @mnquintana. Really hope this will reduce the number of users who are :trollface: by this problem. :pray:

**Comment by mnquintana** [comment]
Added @benogle's examples from #102 now

**Comment by izuzak** [comment]
Great -- thanks! :+1: :boat:

**Pull request merged**